core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/photon/PhotonBroadcastNestedLoopJoinExecParser.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.planparser.photon
+
+import com.nvidia.spark.rapids.tool.planparser.BroadcastNestedLoopJoinExecParserBase
+import com.nvidia.spark.rapids.tool.qualification.PluginTypeChecker
+
+import org.apache.spark.sql.rapids.tool.util.PhotonSparkPlanGraphNode
+
+case class PhotonBroadcastNestedLoopJoinExecParser(
+    node: PhotonSparkPlanGraphNode,
+    checker: PluginTypeChecker,
+    sqlID: Long)
+  extends BroadcastNestedLoopJoinExecParserBase(node, checker, sqlID) {
+
+  override def extractBuildAndJoinTypes(exprStr: String): (String, String) = {
```

Comment:
Is this method the same as `extractBuildAndJoinTypes` in `BroadcastNestedLoopJoinExecParserBase`?

Reply:
The two formats are similar, but the order of arguments is reversed. For example, in:
- `BroadcastNestedLoopJoinExec`, the arguments follow the format:  
  `BuildRight, LeftOuter, ((CEIL(cast(id1#1490 as double)) <= cast(id2#1496 as bigint)))`
- `PhotonBroadcastNestedLoopJoinExec`, the arguments are in the format:  
  `LeftOuter, BuildRight, ((CEIL(cast(id1#1490 as double)) <= cast(id2#1496 as bigint)))`

Due to this Photon-specific argument ordering, we define a custom parser.

Reply:
Thanks for clarifying!